cli/cmd/tap_by_resource.go

```diff
@@ -0,0 +1,182 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/runconduit/conduit/controller/api/util"
+	pb "github.com/runconduit/conduit/controller/gen/public"
+	"github.com/runconduit/conduit/pkg/k8s"
+	"github.com/spf13/cobra"
+)
+
+var (
+	toResource string
+	validArgs  = []string{
+		k8s.KubernetesDeployments,
+		k8s.KubernetesNamespaces,
+		k8s.KubernetesPods,
+		k8s.KubernetesReplicationControllers,
+	}
+)
+
+var tapByResourceCmd = &cobra.Command{
+	Use:   "tapByResource [flags] (RESOURCE)",
+	Short: "Listen to a traffic stream",
+	Long: `Listen to a traffic stream.
+
+  The RESOURCE argument specifies the resource to tap, for example:
+
+  * deploy
+  * deploy/my-deploy
+  * deploy my-deploy
+  * ns/my-ns
+
+  Valid resource types include:
+
+  * deployments
+  * namespaces
+  * pods
+  * replicationcontrollers`,
+	Example: `  # tap the web deployment in the default namespace
+  conduit tapByResource deploy/web
+
+  # tap the web-dlbvj pod in the default namespace
+  conduit tapByResource pod/web-dlbvj
+
+  # tap the test namespace, filter by request to prod namespace
+  conduit tapByResource ns/test --to ns/prod`,
+	Args:      cobra.RangeArgs(1, 2),
+	ValidArgs: validArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		req, err := buildTapByResourceRequest(
+			args, namespace,
+			toResource, toNamespace,
+			maxRps,
+			scheme, method, authority, path,
+		)
+		if err != nil {
+			return err
+		}
+
+		client, err := newPublicAPIClient()
+		if err != nil {
+			return err
+		}
+
+		return requestTapByResourceFromAPI(os.Stdout, client, req)
+	},
+}
+
+func init() {
+	RootCmd.AddCommand(tapByResourceCmd)
+	tapByResourceCmd.PersistentFlags().StringVarP(&namespace, "namespace", "n", "default", "Namespace of the specified resource")
+	tapByResourceCmd.PersistentFlags().StringVar(&toResource, "to", "", "Display requests to this resource")
+	tapByResourceCmd.PersistentFlags().StringVar(&toNamespace, "to-namespace", "", "Sets the namespace used to lookup the \"--to\" resource; by default the current \"--namespace\" is used")
```

Comment:
not a blocker, but would prefer to wrap these very long lines at ~100 chars (i.e. maybe break the descriptions onto their own lines?)